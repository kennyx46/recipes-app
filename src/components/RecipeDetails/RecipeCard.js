import React from 'react';
import ReactMarkdown from 'react-markdown';

import Card from 'react-bootstrap/Card';
import Badge from 'react-bootstrap/Badge';

const RecipeCard = ({ image, title, tags, chef, description }) => {
	return (
		<Card>
		    <Card.Img variant="top" src={image} />
		    <Card.Body>
		        <Card.Title>{title}</Card.Title>
		        <div>
		            {tags.map((tag) => <Badge key={tag} className="mr-2 mb-3" variant="secondary">{tag}</Badge>)}
		        </div>
		      	<ReactMarkdown source={description} />
		    </Card.Body>
		    {chef && <Card.Footer className="font-weight-bold">Chef: {chef}</Card.Footer>}
		</Card>
	);
}

export default RecipeCard;
