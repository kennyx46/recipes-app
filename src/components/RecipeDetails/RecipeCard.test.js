import React from 'react';
import { shallow } from 'enzyme';
import RecipeCard from './RecipeCard';
import Card from 'react-bootstrap/Card';
import Badge from 'react-bootstrap/Badge';

describe('RecipeCard', () => {
    const defaultProps = Object.freeze({
        tags: []
    });

    it('renders component', () => {
        const wrapper = shallow(
            <RecipeCard {...defaultProps }/>
        ).dive();

        expect(wrapper).toMatchSnapshot();
    });

    it('renders recipe details', () => {
        const sampleRecipe = { id: '1', title: 'Awesome food', tags: [] };
        const props = { ...defaultProps, ...sampleRecipe };
        const recipeCard = shallow(<RecipeCard {...props} />);

        expect(recipeCard.find(Card.Title).text()).toContain('Awesome food');
    });

    it('renders recipe tags', () => {
        const sampleRecipe = { id: '1', title: 'Awesome food', tags: ['first Tag', 'second Tag'] };
        const props = { ...defaultProps, ...sampleRecipe };
        const recipeCard = shallow(<RecipeCard {...props} />);

        expect(recipeCard.find(Badge)).toHaveLength(2);
    });

});
