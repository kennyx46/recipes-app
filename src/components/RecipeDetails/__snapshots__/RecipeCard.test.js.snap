// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard renders component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(Bootstrap(Card))>
    <ForwardRef(Bootstrap(CardImg))
      variant="top"
    />
    <ForwardRef(Bootstrap(CardBody))>
      <ForwardRef(Bootstrap(CardTitle)) />
      <div />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        parserOptions={Object {}}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
    </ForwardRef(Bootstrap(CardBody))>
  </ForwardRef(Bootstrap(Card))>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Map {},
        "_currentValue2": Map {},
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Map {},
          "_currentValue2": Map {},
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
}
`;
